### ISSUE TITLE
Improper Versioning Within Code

### ISSUE THREAD

**Issue by lberrymage** [aseprite/aseprite#2175]
### The Problem
To sum, the source code of both the 1.2.15 release zip and the tarball uses the text "1.3-dev" as the version in both [src/config.h](https://github.com/aseprite/aseprite/blob/0fef9bfa6ba24aabc43c85d029f344b398ee478c/src/config.h#L31) and [data/gui.xml](https://github.com/aseprite/aseprite/blob/1c8b8051a881788b849243e210c4f9a2a2763414/data/gui.xml#L5) when, needless to say, it ought to be "1.2.15". Note that the links are to the current master branch, although the files look the same in both the 1.2.15 zip and tarball.

### Aseprite and System version

* Aseprite Version: 1.2.15
* System: Arch Linux Rolling, fully up-to-date as of Oct 11, 2019

**Comment by dacap** [comment]
I've already have a patch to automatically generate the version depending on the current commit using `git describe --tags --dirty=-modified`. The first step was done in a0882ba4435afc6707e780282cc3c5a676d307a1.

I still have my doubts about changing the program version depending on the current commit because it should generate the new version on each compilation depending on any kind of change of the current working directory, and it might not work if the `git` command is not available, or if the source code was downloaded from the .zip (which doesn't include the .git information).

So basically there is no perfect solution for an auto-generated version number. At the moment I think I'll left all compiled versions as `v1.x-dev` and disable the "new version available" warning.

**Comment by andreyorst** [comment]
When building from source, the reported version is unhelpful:

```
$ aseprite --version
Aseprite 1.x-dev
```

i.e., I've rebuilt the latest version, but didn't replace the binary in the `$PATH`, and only noticed later, because of the bug still happening, that was fixed in a later version. I've built from a stable release tag, and I assumed it to have the correct version. Maybe I did something wrong at configuration time, but I followed the instructions in the INSTALL.md.

**Comment by dacap** [comment]
The version of Aseprite is taken from a hard coded string from:

https://github.com/aseprite/aseprite/blob/2ed0f145fbece15875c1f9a206ff248a156a8791/src/ver/CMakeLists.txt#L5

The compilation process doesn't update this string, the string must be manually updated before compiling in case you want a specific version to be reported. Please read [my comment](https://github.com/aseprite/aseprite/issues/2175#issuecomment-599809108).

**Comment by andreyorst** [comment]
Yes, I understand - just was figuring out what happened, and found this ticket. A correct version would help me find the problem faster, that's all.

I made a workaround script:

```sh
SRC_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
BUILD_DIR="$PWD"

GIT_VERSION="$(git -C "$SRC_DIR" describe --tags --always --dirty 2>/dev/null || echo "1.x-dev")"
GIT_VERSION="${GIT_VERSION#v}"

cmake ... \
  -G Ninja "$SRC_DIR"

printf '#define VERSION "%s"\n' "$GIT_VERSION" \
  > "$BUILD_DIR/src/ver/generated_version.h"

ninja "$@"
```

This gives me:

```sh
$ ./aseprite --version
Aseprite 1.3.17.2
```

**Comment by dacap** [comment]
I like the `git describe --tags --always --dirty` solution, I'll try to integrate to the compilation process later.
